### PR TITLE
Some updates to make the plugin work better with inverter SG01HP3

### PR DESCRIPTION
**Pull request by daniel-deptula**
- Improvements for SG01HP3 sensors
- Support for a new sensor (enum-type string values)

### CODE REVIEW

File: plugins/deye_plugin_ha_discovery.py [hunk 1]
```diff
@@ -130,12 +130,12 @@ def _get_device_class(topic: str) -> str:
 
         # topic: ac/l*/voltage
         # topic: dc/pv*/voltage
-        if topic.endswith("/voltage"):
+        if topic.endswith("/voltage") or topic.endswith("/charging_voltage") or topic.endswith("/discharge_voltage"):
```
Comment (CarstenGrohmann) [inline plugins/deye_plugin_ha_discovery.py:133]: Please expand the topics everywhere in the comment above the if statement to simplify later changes.

File: plugins/deye_plugin_ha_discovery.py [hunk 2]
```diff
@@ -258,24 +280,43 @@ def publish_sensor_information(self, topic: str, observation: Observation):
...
+                "state_topic": topic,
+                "device": {
+                    "identifiers": [node_id],
+                    "name": self._device_name,
+                    "manufacturer": self.inverter_manufacturer,
+                    "model": f"{self.inverter_model} SN:{self._logger_serial}",
+                    "serial_number": str(self._logger_serial),
+                    "sw_version": f"deye-inverter-mqtt with {self.get_id()}",
+                },
+            }
```
Comment (CarstenGrohmann) [inline plugins/deye_plugin_ha_discovery.py:319]: What do you think about rollback to the previous code and update the `discover_config` structure in a second step?

Untested example:
```suggestion
        discover_config = {
            "name": observation.sensor.name,
            "unique_id": self._get_unique_id(observation.sensor.name),
            "force_update": True,
            "device_class": device_class,
            "state_class": state_class,
            "unit_of_measurement": self._adapt_unit(observation.sensor.unit),
            "availability_topic": f"{self._config.mqtt.topic_prefix}/status",
            "state_topic": topic,
            "device": {
                "identifiers": [node_id],
                "name": self._device_name,
                "manufacturer": self.inverter_manufacturer,
                "model": f"{self.inverter_model} SN:{self._logger_serial}",
                "serial_number": str(self._logger_serial),
                "sw_version": f"deye-inverter-mqtt with {self.get_id()}",
            },
        }
        if device_class == "enum":
            del discover_config["unit_of_measurement"]
            discover_config["options"] = enum_options
```

**Comment by CarstenGrohmann** [comment]
Please add a small comment to the changelog in the Readme as it is your change and your appreciation.

File: plugins/deye_plugin_ha_discovery.py (hunk 1 above)
Comment (daniel-deptula) [reply plugins/deye_plugin_ha_discovery.py:135]: Done in ce0026a

File: plugins/deye_plugin_ha_discovery.py (hunk 2 above)
Comment (daniel-deptula) [reply plugins/deye_plugin_ha_discovery.py]: I totally agree, that was quick&dirty. Improved in ce0026a (tested).

**Review by CarstenGrohmann — APPROVED** [review]

**Comment by CarstenGrohmann** [comment]
I did some cosmetic changed and merged this PR with f19ddd1. Unfortunatally Github does not detect/accpet this merge commit. I've to close this PR manually.

**Pull request closed without merging**